source/val/validate_type.cpp

```diff
@@ -110,11 +110,10 @@ spv_result_t ValidateTypeArray(ValidationState_t& _, const Instruction* inst) {
   if ((spvIsVulkanEnv(_.context()->target_env) ||
        spvIsWebGPUEnv(_.context()->target_env)) &&
       element_type->opcode() == SpvOpTypeRuntimeArray) {
-    const char* env_text =
-        spvIsVulkanEnv(_.context()->target_env) ? "Vulkan" : "WebGPU";
     return _.diag(SPV_ERROR_INVALID_ID, inst)
            << "OpTypeArray Element Type <id> '" << _.getIdName(element_type_id)
-           << "' is not valid in " << env_text << " environment.";
+           << "' is not valid in "
+           << spvLogStringForEnv(_.context()->target_env) << " environment.";
```

Comment:
The message seems like it might be a bit ackward.  "is not valid in WebGPU environment."  Should it be something like "is not valid in a WebGPU environment" or "is not valid in WebGPU environments"?

Reply:
Prexisting, but I agree is it wrong. Corrected and tests updated.